hikari/config.py

```diff
@@ -261,7 +261,7 @@ def _(self, attrib: attr.Attribute[typing.Optional[float]], value: typing.Option
 
 
 @attr_extensions.with_copy
-@attr.define(kw_only=True, weakref_slot=False)
+@attr.define(kw_only=True, weakref_slot=False, on_setattr=[attr.setters.convert, attr.setters.validate])
```

Comment:
Pretty sure attrs doesn't add `attr.setters.convert` by default, I'll check locally.

Reply:
Confirmed:
```py
In [1]: import attr

In [2]: @attr.define()
   ...: class Example:
   ...:     b: int = attr.ib(converter=int)
   ...: 

In [3]: n = Example(5)

In [4]: n.b = '42'

In [5]: n
Out[5]: Example(b='42')

In [6]: @attr.define(on_setattr=[attr.setters.convert, attr.setters.validate])
   ...: class Example:
   ...:     b: int = attr.ib(converter=int)
   ...: 

In [7]: nn = Example(5)

In [8]: nn.b = '42'

In [9]: nn
Out[9]: Example(b=42)
```

Reply:
Welp that's not the behaviour I'm seeing, on the latest ver of attr i'm getting
```
ok 123
<class 'int'>
```
from the following code
```py
import attr

def test_convert(value: str) -> int:
    print("ok", value)
    return int(value)

@attr.define()
class Foo:
    foo = attr.field(converter=test_convert)


print(type(Foo("123").foo))
```

Also under current behaviour the _ssl field is using a converter without setting the on_setattr and validation or the property's assertion statement would be constantly erroring if the converter wasn't running and it was being left as a boolean value

Reply:
Yeah, converters always work in the initializer. But, the docstring mentions `If assigned to <boolean value>`, which has the connotation of `settings.ssl = False` after making the settings. I may be misreading though!

Reply:
Well typing wise you can't assigned a bool to it anymore cause as far as MYPY's concerned since it's typed as as SSLContext it therefore can only be set as an SSLContext attribute wise although currently this is an unsettable property so that whole "assigned" thing was probably just ambiguous phrasing or outdated docs

Reply:
Also it's probably preferable that people aren't encouraged to set those fields like that since there's no guarantee it'll update the behaviour of whatever client it's attached to so those docs should prob be updated ~~and there might actually be merit in freezing these classes~~

Reply:
> probably just ambiguous phrasing or outdated docs
It's the latter

Reply:
Done